src/Cardano/PoolActions.hs

```diff
@@ -0,0 +1,40 @@
+module Cardano.PoolActions where
+
+import           Cardano.Models
+import           Cardano.Utils
+import           ErgoDex.Contracts.Pool
+import           ErgoDex.Contracts.Types
+import           ErgoDex.OffChain
+import           Plutus.V1.Ledger.Value
+import qualified PlutusTx.AssocMap   as Map
+import           Ledger.Scripts
+import qualified Ledger.Typed.Scripts.Validators as Validators
+import qualified PlutusTx
+import           Data.List
+import           Data.Maybe
+
+createPool :: PoolParams -> FullTxOut -> FullTxOut -> FullTxOut -> TxCandidate
+createPool pp@PoolParams{..} inputWithX inputWithY inputWithNft = let
```

Comment:
`let` should start from a new line

Reply:
What if X, Y are in a single utxo or even X, Y and NFT? Just pass a list of inputs and check that all the required assets are available.

Reply:
Also, what about error cases? Should `createPool` always succeed?